bundles/org.openhab.ui/web/src/assets/definitions/blockly/ohblocks_timers.js

```diff
@@ -0,0 +1,326 @@
+      this.setPreviousStatement(true, null)
+      this.setNextStatement(true, null)
+      this.setColour(230)
+      this.setTooltip('Waits for the specified milliseconds')
+    }
+  }
+
+  /*
+  * sleeps for the number of milliseconds
+  * Code part using Thread.sleep
+  */
+  Blockly.JavaScript['oh_sleep'] = function (block) {
+    const voiceName = Blockly.JavaScript.provideFunction_(
+      'thread',
+      ['var ' + Blockly.JavaScript.FUNCTION_NAME_PLACEHOLDER_ + ' = Java.type(\'java.lang.Thread\')'])
+    let milliseconds = block.getFieldValue('milliseconds')
+
+    let code = `thread.sleep(${milliseconds})\n`
+    return code
+  }
+
+  /*
+  * Simple Timer creation without further control of retriggering, canceling or the like
+  * Blockly part
+  */
+  Blockly.Blocks['oh_simpleTimer'] = {
+    init: function () {
+      this.appendStatementInput('timer')
+        .setCheck(null)
+        .appendField('After')
+        .appendField(new Blockly.FieldNumber(10), 'delay')
+        .appendField(new Blockly.FieldDropdown([['seconds', 'seconds'], ['minutes', 'minutes'], ['hours', 'hours']]), 'time_unit')
+        .appendField('do')
+      this.setPreviousStatement(true, null)
+      this.setNextStatement(true, null)
+      this.setColour(230)
+      this.setTooltip('Simple Timer creation without further control of retriggering, canceling or the like')
+      this.setHelpUrl('https://www.openhab.org/docs/configuration/actions.html#timers')
+    }
+  }
+
+  /*
+  * Simple Timer creation without further control of retriggering, canceling or the like
+  * Code part
+  */
+  Blockly.JavaScript['oh_simpleTimer'] = function (block) {
+    addScriptExecution()
+    addZonedDateTime()
+    let numberDelay = block.getFieldValue('delay')
+    let dropdownTimeUnit = block.getFieldValue('time_unit')
+    let statementsTimer = Blockly.JavaScript.statementToCode(block, 'timer')
+    let unitsFunction = ''
+    switch (dropdownTimeUnit) {
+      case 'seconds' :
+        unitsFunction = 'plusSeconds'
+        break
+      case 'minutes' :
+        unitsFunction = 'plusMinutes'
+        break
+      case 'hours' :
+        unitsFunction = 'plusHours'
+        break
+    }
+    let code = `scriptExecution.createTimer(zonedDateTime.now().${unitsFunction}(${numberDelay}), function(){\n`
+    code += statementsTimer
+    code += '})\n'
+    return code
+  }
+
+  /*
+  * Creates a named timer that starts after the defined delay provided my the number and the unit of time
+  * Blockly part
+  */
+  Blockly.Blocks['oh_namedTimer'] = {
+    init: function () {
+      this.appendStatementInput('runMe')
+        .setCheck(null)
+        .appendField('Run This')
+      this.appendValueInput('delay')
+        .setCheck(null)
+        .appendField('After')
+        .appendField(new Blockly.FieldDropdown([['seconds', 'plusSeconds'], ['minutes', 'plusMinutes'], ['hours', 'plusHours'], ['days', 'plusDays'], ['weeks', 'plusWeeks'], ['months', 'plusMonths']]), 'delayUnits')
```

Comment:
I prefer this one to the timeUnit field of the oh_timer_simple above so perhaps change the implementation of the simple timer to have exactly the same parameter?

Reply:
You mean adding the days, weeks, months to the simpleTimer as they are missing there?

![image](https://user-images.githubusercontent.com/5937600/138288918-37872ce7-a886-4739-884f-33e54b376e72.png)

like so?

![image](https://user-images.githubusercontent.com/5937600/138289074-dd22e2ef-fa94-4563-8d39-cfa087a1ba8a.png)

Reply:
yes have the same options/field name, also avoid the numerical field and use a value input with a Number type constraint instead for the delay (the "10" in the screenshots above), that way you can use a variable or the result of a calculation instead of a predefined value.